### DISCUSSION TITLE
麻雀ゲームのライブラリ

### DISCUSSION THREAD

**Discussion by john-banjirou** [smly/mjai.app#12] [Ideas]
投稿失礼します。

大変すばらしい、試みと存じます。

麻雀ゲームを、自分で作ってみたいと思っております。
そこで、麻雀ゲームのライブラリ？を、公開いただけませんでしょうか？

それとは別に、
当該AI対戦サイトで、人間もエントリーして、戦えるようにしてほしいです。

どこも、信用ならないオンライン麻雀ゲームばかりなのです。

よろしくおねがいします。

**Comment by smly** [comment]
ご意見ありがとうございます！ゲームエンジンは Mortal https://mortal.ekyu.moe/ にインターフェース部分の変更を行ったものを使用しています。この変更を含めたコードは bot 開発者のデバッグ環境を整えるために公開する予定です。

> 人間もエントリーして戦えるように
面白いアイディアありがとうございます。牌山をユーザーが定義したり、様々な遊び方ができそうです。
一方で「アップロードしたモデルは本サイトで他プレイヤーと対戦する目的のみに使用されます」と利用範囲を明確にしています。
もし機能実装をする場合は、改めて bot 開発者に同意を得た上で行うことになりそうです。
いまのところ麻雀AI の開発意欲のほうが強いため機能実装する予定はありません。時間が許するならば実装して遊びたくあります。
